### PR TITLE
Java to csharp code conversion

### PR DESCRIPTION
**Pull request by iamguru53**
I have done below changes:
- Moved java source code to seperate folder
- Created project to converted java code to c#

### CODE REVIEW

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Department.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Abstraction
+{
+    public class Department
+    {
+        private String id;
+        private String name;
+        public String GetName()
```
Comment (emperorsreeni) [inline TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Department.cs]: Convert all getters and setters to auto properties in all files

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Employee.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Abstraction
+{
+    public class Employee
+    {
```
Comment (emperorsreeni) [inline TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Employee.cs:8]: Use an appropriate naming convention for private variables. Use underscore.

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Employee.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        private String name;
+        private String id;
+        private double salary;
+        private String departmentName;
+        public Employee(String name, String id, double salary, String departmentName)
+        {
+            this.name = name;
+            this.id = id;
```
Comment (emperorsreeni) [inline TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Employee.cs]: remove **this** keywords everywhere

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Program.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+using telldonotask.Good;
+
+
+namespace OOPrinciples
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            #region Abstraction
```
Comment (emperorsreeni) [inline TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Program.cs]: Remove regions. It is not recommended the practice to use regions

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/TellDonotAsk/Bad/Unit.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace telldonotask.Bad
```
Comment (emperorsreeni) [inline TDD Boot Camp/Part 01/Code/C#/OOPrinciples/TellDonotAsk/Bad/Unit.cs]: Use a proper naming convention for namespace

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/bin/Debug/OOPrinciples.exe.config [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (emperorsreeni) [inline TDD Boot Camp/Part 01/Code/C#/OOPrinciples/bin/Debug/OOPrinciples.exe.config]: Remove bin, obj folder and update .ignore file to exclude these folders

**Review by emperorsreeni — CHANGES_REQUESTED** [review]
**Quick summary:**
1. Convert all setters and getters to auto-properties
2. Use the appropriate naming convention for private variables
3. Remove **this** keywords
4. Use the proper naming convention for the namespace in some of the file

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Employee.cs (hunk 2 above)
Comment (iamguru53) [reply TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Employee.cs:8]: done

**Pull request merged**